test/over_react/component_declaration/builder_integration_tests/new_boilerplate/stateful_component_integration_test.dart

```diff
@@ -0,0 +1,135 @@
+// Copyright 2016 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:over_react/over_react.dart';
+import 'package:test/test.dart';
+
+import '../../../../test_util/test_util.dart';
+
+part 'stateful_component_integration_test.over_react.g.dart';
+
+main() {
+  group('(New boilerplate) stateful component integration:', () {
+    test('component cannot set state directly in init', () {
+      expect(() => render((StatefulComponentTest()..setStateDirectly = true)()),
+          throwsA(hasToStringValue('Assertion failed: "Component2.state should '
+              'only be set via initializeState (within the init lifecycle method) '
+              'or setState."')));
+    }, testOn: '!js');
```

Comment:
Since we're here 🤷‍♂ 
```suggestion
    }, tags: 'no-dart2js');
```